webapp/components/generic/SearchableInput/SearchableInput.vue

```diff
@@ -0,0 +1,227 @@
+<template>
+  <div
+    class="searchable-input"
+    aria-label="search"
+    role="search"
+    :class="{
+      'is-active': isActive,
+      'is-open': isOpen,
+    }"
+  >
+    <div class="field">
+      <div class="control">
+        <ds-button v-if="isActive" icon="close" ghost class="search-clear-btn" @click="clear" />
+        <ds-select
+          type="search"
+          icon="search"
+          v-model="searchValue"
+          :id="id"
+          label-prop="id"
+          :icon-right="isActive ? 'close' : null"
+          :options="options"
+          :loading="loading"
+          :filter="item => item"
+          :no-options-available="emptyText"
+          :auto-reset-search="!searchValue"
+          :placeholder="$t('search.placeholder')"
+          @click.capture.native="isOpen = true"
+          @focus.capture.native="onFocus"
+          @input.native="handleInput"
+          @keyup.enter.native="onEnter"
+          @keyup.delete.native="onDelete"
+          @keyup.esc.native="clear"
+          @blur.capture.native="onBlur"
+          @input.exact="onSelect"
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-01-08T12:21:27Z" title="Wednesday, January 8th 2020, 1:21:27 pm +01:00">Jan 8, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,236 @@
+<template>
+  <div
+    class="searchable-input"
+    aria-label="search"
+    role="search"
+    :class="{
+      'is-active': isActive,
+      'is-open': isOpen,
+    }"
+  >
+    <div class="field">
+      <div class="control">
+        <ds-button v-if="isActive" icon="close" ghost class="search-clear-btn" @click="clear" />
+        <ds-select
+          type="search"
+          icon="search"
+          v-model="searchValue"
+          :id="id"
+          label-prop="id"
+          :icon-right="isActive ? 'close' : null"
+          :options="options"
+          :loading="loading"
+          :filter="item => item"
+          :no-options-available="emptyText"
+          :auto-reset-search="!searchValue"
+          :placeholder="$t('search.placeholder')"
+          @click.capture.native="isOpen = true"
+          @focus.capture.native="onFocus"
+          @input.native="handleInput"
+          @keyup.enter.native="onEnter"
+          @keyup.delete.native="onDelete"
+          @keyup.esc.native="clear"
+          @blur.capture.native="onBlur"
+          @input.exact="onSelect"
```


Whoa, a lot of things going on here – do we really need all these props? I guess the `icon-right` doesn't do anything anymore, since we have the button on top of it now. Anything else that's maybe just left over from before?

I'm also wondering if some of these methods could be combined into one to make it more readable – does `onFocus` really have a different impact than `onClick`? Should different things happen `onBlur` and on `escape`? And isn't `ds-select` handling e.g. `onDelete` by itself already? You'll know better @Mogge, so if you say it all makes sense as it is I trust you! Just saying it's not really easy to follow and I'd be glad if it could be "cleaned up" a bit. 😅

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-10T10:24:59Z" title="Friday, January 10th 2020, 11:24:59 am +01:00">Jan 10, 2020</time>_

---


hey @alina-beck @Mogge and I are pairing on this `PR` review change requests and we agree to leave this for now, as 1) this functionality is copy and pasted from what is already in production, and 2) we will refactor this further when we migrate `Select` to our repo.

the difference between `onFocus` and `@click` is just that `onFocus` calls `clearTimeout`... we could discuss whether this is needed, but certainly not something to block it getting merged in, I think you'd agree?

the difference between `onBlur` and `esc` is that `onBlur does not clear the user's search term because they can accidentally click somewhere besides the search field and then be annoyed at losing their search term. Whereas, if a user types `Esc`, I think it's reasonable to assume they want to clear their search. What do you think?